### PR TITLE
Deprecates  in hub.py

### PR DESCRIPTION
**Pull request by jdsgomes**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67914

Adresses #67391 

cc @nairbv @NicolasHug @vmoens @jdsgomes @albanD

### CODE REVIEW

File: torch/hub.py
Comment (albanD) [inline torch/hub.py:71]: nit you can make it a `DeprecationWarning` as well.

**Review by albanD — APPROVED** [review]
Thanks for the update!

File: torch/hub.py
Comment (NicolasHug) [inline torch/hub.py:71]: Let's be very explicit here to help users figure out the timeline:


```suggestion
    warnings.warn('The use of torch.hub.import_module is deprecated in v0.11 and will be removed in v0.12')
```

**Review by NicolasHug — APPROVED** [review]
Thanks @jdsgomes ! LGTM with a minor comment.

After this is landed, would you mind submitting PRs to the repos mentioned in https://github.com/pytorch/pytorch/issues/67391#issuecomment-953612004 to update their code?

**Comment by jdsgomes** [comment]
> Thanks @jdsgomes ! LGTM with a minor comment.
> 
> After this is landed, would you mind submitting PRs to the repos mentioned in [#67391 (comment)](https://github.com/pytorch/pytorch/issues/67391#issuecomment-953612004) to update their code?

Sure, I will also work on those changes. Just waiting for some approvals to have access to the repos

File: torch/hub.py
Comment (NicolasHug) [inline torch/hub.py:72]: I missed this bit earlier, we should also preserve the return value:

```suggestion
    return _import_module()
```

**Pull request merged**
